File: blue-serial/src/main/java/io/github/chayanforyou/blueserial/le/SerialListener.java
package io.github.chayanforyou.blueserial.le;

public interface SerialListener {
    void onSerialConnect();

    void onSerialConnectError(Exception e);

    void onSerialRead(byte[] data);                // socket -> service

    void onSerialIoError(Exception e);
}

File: blue-serial/src/main/java/io/github/chayanforyou/blueserial/le/SerialListener.kt
package io.github.chayanforyou.blueserial.le

interface SerialListener {
    fun onSerialConnect()
    fun onSerialConnectError(e: Exception)
    fun onSerialRead(data: ByteArray) // socket -> service
    fun onSerialIoError(e: Exception)
}
